apps/daisy/settings.js

```diff
@@ -6,53 +6,95 @@
+           'idxInfo' : 0,
+           'step_target' : 10000};
 
   // ...and overwrite them with any saved values
   // This way saved values are preserved if a new version adds more settings
-  const storage = require('Storage')
+  const storage = require('Storage');
   let settings = storage.readJSON(SETTINGS_FILE, 1) || s;
-  const saved = settings || {}
+  const saved = settings || {};
   for (const key in saved) {
-    s[key] = saved[key]
+    s[key] = saved[key];
   }
 
   function save() {
-    settings = s
-    storage.write(SETTINGS_FILE, settings)
+    settings = s;
+    storage.write(SETTINGS_FILE, settings);
   }
 
   var color_options = ['Green','Orange','Cyan','Purple','Red','Blue'];
   var fg_code = ['#0f0','#ff0','#0ff','#f0f','#f00','#00f'];
   var gy_code = ['#020','#220','#022','#202','#200','#002'];
-  
-  E.showMenu({
-    '': { 'title': 'Daisy Clock' },
-    '< Back': back,
-    'Colour': {
-      value: 0 | color_options.indexOf(s.color),
-      min: 0, max: 5,
-      format: v => color_options[v],
-      onchange: v => {
-        s.color = color_options[v];
-        s.fg = fg_code[v];
-        s.gy = gy_code[v];
-        save();
+  var ring_options = ['Hours', 'Minutes', 'Seconds', 'Steps', 'Battery', 'Sun'];
+  var step_options = [100, 1000, 5000, 10000, 15000, 20000];
+
+  function showMainMenu() {
+    let appMenu = {
+      '': { 'title': 'Daisy Clock' },
+      '< Back': back,
+      'Colour': {
+        value: 0 | color_options.indexOf(s.color),
+        min: 0, max: color_options.length - 1,
+        format: v => color_options[v],
+        onchange: v => {
+          s.color = color_options[v];
+          s.fg = fg_code[v];
+          s.gy = gy_code[v];
+          save();
+        },
+      },
+      'Idle Warning': {
+        value: !!s.idle_check,
+        onchange: v => {
+          s.idle_check = v;
+          save();
+        },
       },
-    },
-    'Idle Warning': {
-      value: !!s.idle_check,
-      onchange: v => {
-        s.idle_check = v;
-        save();
+      'Expected Battery Life In Days Not Percentage': {
+        value: !!s.batt_hours,
+        onchange: v => {
+          s.batt_hours = v;
+          save();
+        },
       },
-    },
-    'Expected Battery Life In Days Not Percentage': {
-      value: !!s.batt_hours,
-      onchange: v => {
-        s.batt_hours = v;
-        save();
+      '12 Hr Time': {
+        value: !!s.hr_12,
+        onchange: v => {
+          s.hr_12 = v;
+          save();
+        },
       },
-    }
-  });
-})
+      'Ring Display': {
+        value: 0 | ring_options.indexOf(s.ring),
+        min: 0, max: ring_options.length - 1,
+        format: v => ring_options[v],
+        onchange: v => {
+          let prev = s.ring;
+          s.ring = ring_options[v];
+          save();
+          if (prev != s.ring && (prev === 'Steps' || s.ring === 'Steps')) {
+            setTimeout(showMainMenu, 0);  // Reference https://github.com/orgs/espruino/discussions/7697
```

Comment:
Perhaps we inline the reason here too, wdyt?

```suggestion
            // redisplay the menu with/without ring setting
            // Reference https://github.com/orgs/espruino/discussions/7697
            setTimeout(showMainMenu, 0);
```

Reply:
I like it. I'll add it into my next commit

Reply:
Changed [here](https://github.com/espruino/BangleApps/pull/3759/commits/f431258eafca3167610b295b625fda24fa87ea38).

Reply:
Smashing!